bin/generate-php-sync-issue.mjs

```diff
@@ -0,0 +1,460 @@
+		);
+		process.exit( 1 );
+	}
+
+	console.log( 'Welcome to the PHP Sync Issue Generator!' );
+	const octokit = new Octokit( {
+		auth: authToken,
+	} );
+
+	console.log( '--------------------------------' );
+	console.log( '• Running script...' );
+
+	// These should be paths where we expect to find PHP files that
+	// will require syncing to WordPress Core. This list should be
+	// extremely selective.
+	const paths = [ '/lib', '/packages/block-library', '/phpunit' ];
+
+	console.log( `• Fetching all commits made to ${ REPO } since: ${ since }` );
+	let commits = await getAllCommitsFromPaths(
+		octokit,
+		OWNER,
+		REPO,
+		since,
+		paths
+	);
+
+	// Remove identical commits based on sha
+	commits = commits.reduce( ( acc, current ) => {
+		const x = acc.find( ( item ) => item.sha === current.sha );
+		if ( ! x ) {
+			return acc.concat( [ current ] );
+		}
+		return acc;
+	}, [] );
+
+	// Fetch the full commit data for each of the commits.
+	// This is because the /commits endpoint does not include the
+	// information about the `files` modified in the commit.
+	console.log(
+		`• Fetching full commit data for ${ commits.length } commits`
+	);
+	const commitsWithCommitData = await Promise.all(
+		commits.map( async ( commit ) => {
+			const commitData = await getCommit( octokit, commit.sha );
+
+			// Our Issue links to the PRs associated with the commits so we must
+			// provide this data. We could also get the PR data from the commit data,
+			// using getPullRequestDataForCommit, but that requires yet another
+			// network request. Therefore we optimise for trying to build
+			// the PR URL from the commit data we have available.
+			const pullRequest = {
+				url: buildPRURL( commit ),
+				creator: commit?.author?.login || 'unknown',
+			};
+
+			if ( pullRequest ) {
+				commitData.pullRequest = pullRequest;
+			}
+
+			return commitData;
+		} )
+	);
+
+	const processResult = pipe(
+		processCommits,
+		removeNesting,
+		removeSinglePRLevels,
+		dedupePRsPerLevel,
+		removeEmptyLevels,
+		sortLevels
+	);
+
+	console.log( `• Processing ${ commitsWithCommitData.length } commits` );
+	const result = processResult( commitsWithCommitData );
+
+	console.log( `• Generating Issue content` );
+	const content = generateIssueContent( result );
+
+	// Write the Markdown content to a file
+	fs.writeFileSync( nodePath.join( __dirname, 'issueContent.md' ), content );
+}
+
+async function getAllCommitsFromPaths( octokit, owner, repo, since, paths ) {
+	let commits = [];
+
+	for ( const path of paths ) {
+		const pathCommits = await getAllCommits(
+			octokit,
+			owner,
+			repo,
+			since,
+			path
+		);
+		commits = [ ...commits, ...pathCommits ];
+	}
+
+	return commits;
+}
+
+function buildPRURL( commit ) {
+	const prIdMatch = commit.commit.message.match( /\(#(\d+)\)/ );
+	const prId = prIdMatch ? prIdMatch[ 1 ] : null;
+	return prId
+		? `https://github.com/WordPress/gutenberg/pull/${ prId }`
+		: `[Commit](${ commit.html_url })`;
```

Comment:
If we can't build the PR URL from the commit message, then we'll fall back to the commit URL itself and let GH UI show the link to the PR there.